doc(kuma-cp) Zone Ingress (#458)
kumahq/kuma-website: docs/docs/1.2.0/security/certificates.md

:::
::::

### Zone Ingress proxy authentication

> To obtain an Envoy resources over xDS from the server, a zone-ingress proxy must authenticate itself.

Reviewer: ```suggestion
To obtain Envoy resources over xDS from the server, a zone-ingress proxy must authenticate itself.
```

:::: tabs :options="{ useUrlFragment: false }"

::: tab "Kubernetes (Service Account Token)"

> A zone-ingress proxy proves its identity by leveraging [ServiceAccountToken](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/#service-account-automation) that is mounted in every pod.

Reviewer: ```suggestion
A zone-ingress proxy proves its identity by leveraging [the ServiceAccountToken](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/#service-account-automation) that is mounted in every pod.
```

:::

::: tab "Universal (Zone Ingress proxy token)"
A zone-ingress proxy must be explicitly configured with a unique security token (Zone Ingress proxy token) that will be used to prove its identity. 
The Zone Ingress proxy token is a signed [JWT token](https://jwt.io) that carries the zone-ingress proxy name and name of the zone that it belongs to.

> It is signed by an RSA key auto-generated by the control plane on the first start. Tokens are never stored in the control plane,
> the only thing that is stored is a signing key that is used to verify if a token is valid.

Reviewer: ```suggestion
It is signed by an RSA key that is generated by the control plane on first start. Tokens are never stored in the control plane -- only a signing key is stored, to verify whether a token is valid.
```

> You can generate token either by REST API

Reviewer: ```suggestion
Generate the token with the REST API:
```

```bash
$ curl -XPOST \ 
  -H "Content-Type: application/json" \
  --data '{"zone": "us-east"}' \
  http://localhost:5681/tokens/zone-ingress
```

> or by using `kumactl`

Reviewer: ```suggestion
or with `kumactl`:
```

```bash
$ kumactl generate zone-ingress-token \
  --zone us-east  > /tmp/kuma-dp-echo1-token
``` 

> The token should be stored in a file and then used when starting `kuma-dp`

Reviewer: ```suggestion
The token should be stored in a file and then passed when you start `kuma-dp`:
```

```bash
$ kuma-dp run \
  --proxy-type=ingress \
  --name=dp-ingress \
  --cp-address=https://127.0.0.1:5678 \
  --dataplane-token-file=/tmp/kuma-dp-echo-1-token
```

> You can also pass Zone Ingress Token as `KUMA_DATAPLANE_RUNTIME_TOKEN` Environment Variable.

Reviewer: ```suggestion
You can also pass the token as a `KUMA_DATAPLANE_RUNTIME_TOKEN` environment variable.
```

:::

::: tab "Off"

> We can turn off the authentication by setting `KUMA_DP_SERVER_AUTH_TYPE` to `none`.

Reviewer: ```suggestion
You can turn off authentication by setting `KUMA_DP_SERVER_AUTH_TYPE` to `none`.
```

::: warning

> If we disable the authentication between the control plane and the data plane proxies, any data plane proxy will be able to impersonate any service, therefore this is not recommended in production.

Reviewer: ```suggestion
You should not disable authentication between the control plane and the data plane proxies in production. Disabling means that any data plane proxy can impersonate any service.
```
